transport/src/test/java/io/netty/channel/pool/FixedChannelPoolTest.java

```diff
@@ -297,6 +305,12 @@ public void run() {
         pool.release(channel).syncUninterruptibly();
         sc.close().syncUninterruptibly();
         channel.close().syncUninterruptibly();
+
+        ChannelPool.Stats stats = pool.stats();
+        assertEquals(0, stats.leasedChannelsCount());
+        assertEquals(0, stats.pendingChannelsCount());
+        // TODO: (need to wait the close operation is completed. how?
+//        assertEquals(0, stats.availableChannelsCount());
```

Comment:
Almost every `close()` methods in Netty are asynchronous IIRC. Why don't we just let it return a future?